src/pages/SubscriptionService.tsx

```diff
@@ -83,17 +83,16 @@ class SubscriptionService extends Component<Props, SubscriptionServiceState> {
     const {
       target: { name, value }
     } = event;
+    const inputValue = inputValueMap(name, value);
     Service.decode({
       ...this.state.service,
       service_metadata: {
         ...(this.state.service && this.state.service.service_metadata
           ? this.state.service.service_metadata
           : undefined),
-        [name]: inputValueMap(name, value)
+        [name]: inputValue === "" ? undefined : inputValue
```

Comment:
`NonEmptyString` already considers other nullity cases

```suggestion 
        [name]: NonEmptyString.decode(inputValue).getOrElse(undefined)
```

Reply:
The argument of `getOrElse` can not be `undefined`

Reply:
ok, my bad, we cannot use `NonEmptyString`. I'm just not completely sure ` === ""` covers all possible values. If you are, just ignore this comment.